dubbo-remoting/dubbo-remoting-api/src/main/java/org/apache/dubbo/remoting/exchange/support/DefaultFuture.java

```diff
@@ -203,7 +204,9 @@ private void doReceived(Response res) {
             this.complete(res.getResult());
         } else if (res.getStatus() == Response.CLIENT_TIMEOUT || res.getStatus() == Response.SERVER_TIMEOUT) {
             this.completeExceptionally(new TimeoutException(res.getStatus() == Response.SERVER_TIMEOUT, channel, res.getErrorMessage()));
-        } else {
+        } else if(res.getStatus() == Response.SERIALIZATION_ERROR){
+            this.completeExceptionally(new SerializationException(channel, res.getErrorMessage()));
```

Comment:
`new RpcException(SERIALIZATION_EXCEPTION, res.getErrorMessaeg())` would be enough?

Reply:
As shown in the figure below, in the invoke method of AsyncToSyncInvoker, other exceptions are uniformly converted to RpcException. If you directly create a new RpcException(SERIALIZATION_EXCEPTION, res.getErrorMessage()) and throw it at the place you mentioned, would it conflict with the existing code logic?

Also, why did building and testing fail without any error messages?

![image](https://github.com/apache/dubbo/assets/8465969/ebf9a3c9-0c33-4007-92f8-f94e3d9d25cf)